spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -850,4 +865,46 @@ public String unknown(String sourceName, int sourceId, String transform,
       return String.format("%s(%s) %s %s", transform, sourceName, direction, nullOrder);
     }
   }
+
+  public static Object convertPartitionType(Object value, DataType dataType) {
+    if (value == null && dataType instanceof NullType) {
+      return null;
+    }
+    String old = String.valueOf(value);
+    if (dataType instanceof BooleanType) {
+      return DatatypeConverter.parseBoolean(old);
+    }
+    else if (dataType instanceof ByteType) {
```

Comment:
The code style should be consistent with the existing code:
```java
if (...) {
  ...
} else if (...) {
  ...
}
```

Reply:
```
if (...) {
  ...
} else if (...) {
  ...
}

or

if (...) {
  ...
} else if (...) {
  ...
}
else{
  ...
}
```
above two code styles are well

Reply:
added this code, thanks
```
} else {
      return value;
    }
```